src/Compilers/CSharp/Portable/Symbols/Symbol.cs

```diff
@@ -346,6 +346,36 @@ internal virtual LexicalSortKey GetLexicalSortKey()
         public Location GetFirstLocation()
             => TryGetFirstLocation() ?? throw new InvalidOperationException("Symbol has no locations");
 
+        /// <summary>
+        /// Determines if there is a location (see <see cref="Locations"/>) for this symbol whose span is in <paramref
+        /// name="tree"/> and is contained within <paramref name="declarationSpan"/>.  Subclasses can override this to
+        /// be more efficient if desired (especially if avoiding allocations of the <see cref="Locations"/> array is
+        /// desired).
+        /// </summary>
+        public virtual bool HasLocationContainedWithin(SyntaxTree tree, TextSpan declarationSpan, out bool wasZeroWidthMatch)
```

Comment:
>declarationSpan [](http://example.com/codeflow?start=81&length=15)

This isn't necessarily a declarationSpan, right?

Reply:
i'm just using the original name the caller was using :)